dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java

```diff
@@ -64,16 +102,42 @@ public final String eval() {
             serve(out);
             return new String(out.toByteArray());
         } catch (DotDataException | IOException | DotSecurityException e) {
+            Logger.debug(VelocityModeHandler.class, e.getMessage(), e);
             throw new DotRuntimeException(e);
         }
     }
-    
-    public static final VelocityModeHandler modeHandler(PageMode mode, HttpServletRequest request, HttpServletResponse response, String uri, Host host) {
-        return pageModeVelocityMap.get(mode).apply(request, response, uri, host);
+
+    public static final VelocityModeHandler modeHandler(final PageMode mode, final HttpServletRequest request, final HttpServletResponse response, final String uri, final Host host) {
+        // Find the current language
+        final IHTMLPage htmlPage= getHtmlPageFromURI(mode, request, response, uri, host);
+        return pageModeVelocityMap.get(mode).apply(request, response, htmlPage, host);
     }
-    
-    public static final VelocityModeHandler modeHandler(PageMode mode, HttpServletRequest request, HttpServletResponse response) {
-        return pageModeVelocityMap.get(mode).apply(request, response, request.getRequestURI(), hostWebAPI.getCurrentHostNoThrow(request));
+
+    protected static IHTMLPage getHtmlPageFromURI(PageMode mode, HttpServletRequest request, HttpServletResponse response, String uri, Host host) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'uri' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4842460)